kotlinx-collections-immutable/src/main/kotlin/kotlinx/collections/immutable/extensions.kt

```diff
@@ -29,140 +29,526 @@ import kotlinx.collections.immutable.implementations.persistentOrderedSet.Persis
+
+/**
+ * Returns a new persistent map with the specified contents, given as a list of pairs
+ * where the first component is the key and the second is the value.
+ *
+ * If multiple pairs have the same key, the resulting map will contain the value from the last of those pairs.
+ *
+ * Order of the entries in the returned map is unspecified.
+ */
 fun <K, V> persistentHashMapOf(vararg pairs: Pair<K, V>): PersistentMap<K, V> = PersistentHashMap.emptyOf<K,V>().mutate { it += pairs }
 
+
+/**
+ * Returns a new persistent list of the specified elements.
+ */
 @Deprecated("Use persistentListOf instead.", ReplaceWith("persistentListOf(*elements)"))
 fun <E> immutableListOf(vararg elements: E): PersistentList<E> = persistentListOf(*elements)
+
+/**
+ * Returns an empty persistent list.
+ */
 @Deprecated("Use persistentListOf instead.", ReplaceWith("persistentListOf()"))
 fun <E> immutableListOf(): PersistentList<E> = persistentListOf()
 
+
+/**
+ * Returns a new persistent set with the given elements.
+ *
+ * Elements of the returned set are iterated in the order they were specified.
+ */
 @Deprecated("Use persistentSetOf instead.", ReplaceWith("persistentSetOf(*elements)"))
 fun <E> immutableSetOf(vararg elements: E): PersistentSet<E> = persistentSetOf(*elements)
+
+/**
+ * Returns an empty persistent set.
+ */
 @Deprecated("Use persistentSetOf instead.", ReplaceWith("persistentSetOf()"))
 fun <E> immutableSetOf(): PersistentSet<E> = persistentSetOf()
 
+
+/**
+ * Returns a new persistent set with the given elements.
+ *
+ * Order of the elements in the returned set is unspecified.
+ */
 @Deprecated("Use persistentHashSetOf instead.", ReplaceWith("persistentHashSetOf(*elements)"))
 fun <E> immutableHashSetOf(vararg elements: E): PersistentSet<E> = persistentHashSetOf(*elements)
 
+
+/**
+ * Returns a new persistent map with the specified contents, given as a list of pairs
+ * where the first component is the key and the second is the value.
+ *
+ * If multiple pairs have the same key, the resulting map will contain the value from the last of those pairs.
+ *
+ * Entries of the map are iterated in the order they were specified.
+ */
 @Deprecated("Use persistentMapOf instead.", ReplaceWith("persistentMapOf(*pairs)"))
 fun <K, V> immutableMapOf(vararg pairs: Pair<K, V>): PersistentMap<K, V> = persistentMapOf(*pairs)
+
+/**
+ * Returns a new persistent map with the specified contents, given as a list of pairs
+ * where the first component is the key and the second is the value.
+ *
+ * If multiple pairs have the same key, the resulting map will contain the value from the last of those pairs.
+ *
+ * Order of the entries in the returned map is unspecified.
+ */
 @Deprecated("Use persistentHashMapOf instead.", ReplaceWith("persistentHashMapOf(*pairs)"))
 fun <K, V> immutableHashMapOf(vararg pairs: Pair<K, V>): PersistentMap<K, V> = persistentHashMapOf(*pairs)
 
 
-
+/**
+ * Returns an immutable list containing all elements of this collection.
+ */
```

Comment:
Makes sense to add note that it returns the same list if it's already immutable, see the proposal for example.

Reply:
done